### PR TITLE
[1233] Handle Form Description Widgets styles

### PR DESCRIPTION
**Pull request by AxelRICHARD**
# Pull request template

Handle the styling of the Select, MultiSelect, Textarea, Radio, Checkbox widgets from View DSL.

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [x] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [x] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [x] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [x] Variables have a proper type
- [x] Functions’ arguments have a proper type
- [x] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [x] All components have a proper typing for their props
- [x] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [x] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [x] Are data structure classes properly separated from behavioral classes?
- [x] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [x] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by gcoutable — CHANGES_REQUESTED** [review]

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/CheckboxStyleProvider.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * The style provider for the Checkbox Description widget of the View DSL.
+ *
+ * @author arichard
+ */
+public class CheckboxStyleProvider implements Function<VariableManager, CheckboxStyle> {
+
+    private final org.eclipse.sirius.components.view.CheckboxDescriptionStyle viewStyle;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/CheckboxStyleProvider.java]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/MultiSelectStyleProvider.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * The style provider for the MultiSelect Description widget of the View DSL.
+ *
+ * @author arichard
+ */
+public class MultiSelectStyleProvider implements Function<VariableManager, MultiSelectStyle> {
+
+    private final org.eclipse.sirius.components.view.MultiSelectDescriptionStyle viewStyle;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/MultiSelectStyleProvider.java]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/MultiSelectStyleProvider.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    public MultiSelectStyleProvider(org.eclipse.sirius.components.view.MultiSelectDescriptionStyle viewStyle) {
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/MultiSelectStyleProvider.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/CheckboxElementProps.java [hunk 4]
```diff
@@ -112,6 +121,11 @@ public Builder newValueHandler(Function<Boolean, IStatus> newValueHandler) {
             return this;
         }
 
+        public Builder style(CheckboxStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/CheckboxElementProps.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/MultiSelectElementProps.java [hunk 5]
```diff
@@ -128,6 +137,11 @@ public Builder newValuesHandler(Function<List<String>, IStatus> newValuesHandler
             return this;
         }
 
+        public Builder style(MultiSelectStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/MultiSelectElementProps.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/RadioElementProps.java [hunk 6]
```diff
@@ -113,6 +122,11 @@ public Builder newValueHandler(Function<String, IStatus> newValueHandler) {
             return this;
         }
 
+        public Builder style(RadioStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/RadioElementProps.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/TextareaElementProps.java [hunk 7]
```diff
@@ -112,6 +121,11 @@ public Builder newValueHandler(Function<String, IStatus> handler) {
             return this;
         }
 
+        public Builder style(TextareaStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/TextareaElementProps.java]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/SelectStyleProvider.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * The style provider for the Select Description widget of the View DSL.
+ *
+ * @author arichard
+ */
+public class SelectStyleProvider implements Function<VariableManager, SelectStyle> {
+
+    private final org.eclipse.sirius.components.view.SelectDescriptionStyle viewStyle;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/SelectStyleProvider.java]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/TextareaStyleProvider.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * The style provider for the Textarea Description widget of the View DSL.
+ *
+ * @author arichard
+ */
+public class TextareaStyleProvider implements Function<VariableManager, TextareaStyle> {
+
+    private final org.eclipse.sirius.components.view.TextareaDescriptionStyle viewStyle;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/TextareaStyleProvider.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Checkbox.java [hunk 10]
```diff
@@ -96,6 +104,11 @@ public Builder newValueHandler(Function<Boolean, IStatus> handler) {
             return this;
         }
 
+        public Builder style(CheckboxStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Checkbox.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/CheckboxStyle.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public static final class Builder {
+        private String color;
+
+        private Builder() {
+        }
+
+        public Builder color(String color) {
+            this.color = color;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/CheckboxStyle.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelect.java [hunk 12]
```diff
@@ -110,6 +118,11 @@ public Builder newValuesHandler(Function<List<String>, IStatus> newValuesHandler
             return this;
         }
 
+        public Builder style(MultiSelectStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelect.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelectStyle.java [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+
+        private boolean underline;
+
+        private boolean strikeThrough;
+
+        private Builder() {
+        }
+
+        public Builder backgroundColor(String backgroundColor) {
+            this.backgroundColor = backgroundColor;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelectStyle.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelectStyle.java [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+            return this;
+        }
+
+        public Builder foregroundColor(String foregroundColor) {
+            this.foregroundColor = foregroundColor;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/MultiSelectStyle.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Radio.java [hunk 15]
```diff
@@ -96,6 +104,11 @@ public Builder newValueHandler(Function<String, IStatus> newValueHandler) {
             return this;
         }
 
+        public Builder style(RadioStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Radio.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/RadioStyle.java [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+        private boolean underline;
+
+        private boolean strikeThrough;
+
+        private Builder() {
+        }
+
+        public Builder color(String color) {
+            this.color = color;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/RadioStyle.java]: done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Select.java [hunk 17]
```diff
@@ -109,6 +117,11 @@ public Builder newValueHandler(Function<String, IStatus> newValueHandler) {
             return this;
         }
 
+        public Builder style(SelectStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Select.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/SelectStyle.java [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+
+        private boolean underline;
+
+        private boolean strikeThrough;
+
+        private Builder() {
+        }
+
+        public Builder backgroundColor(String backgroundColor) {
+            this.backgroundColor = backgroundColor;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/SelectStyle.java:78]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Textarea.java [hunk 19]
```diff
@@ -96,6 +104,11 @@ public Builder newValueHandler(Function<String, IStatus> newValueHandler) {
             return this;
         }
 
+        public Builder style(TextareaStyle style) {
+            this.style = style;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Textarea.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/TextareaStyle.java [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+
+        private boolean underline;
+
+        private boolean strikeThrough;
+
+        private Builder() {
+        }
+
+        public Builder backgroundColor(String backgroundColor) {
+            this.backgroundColor = backgroundColor;
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/TextareaStyle.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/CheckboxComponent.java [hunk 21]
```diff
@@ -49,6 +50,11 @@ public Element render() {
         Boolean value = checkboxDescription.getValueProvider().apply(variableManager);
         BiFunction<VariableManager, Boolean, IStatus> genericHandler = checkboxDescription.getNewValueHandler();
         Function<Boolean, IStatus> specializedHandler = newValue -> genericHandler.apply(variableManager, newValue);
+        Function<VariableManager, CheckboxStyle> styleProvider = checkboxDescription.getStyleProvider();
+        CheckboxStyle checkboxStyle = null;
+        if (styleProvider != null) {
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/CheckboxComponent.java]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/MultiSelectComponent.java [hunk 22]
```diff
@@ -74,13 +75,19 @@ public Element render() {
         Function<List<String>, IStatus> newValuesHandler = newValues -> {
             return multiSelectDescription.getNewValuesHandler().apply(variableManager, newValues);
         };
+        Function<VariableManager, MultiSelectStyle> styleProvider = multiSelectDescription.getStyleProvider();
+        MultiSelectStyle multiSelectStyle = null;
+        if (styleProvider != null) {
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/MultiSelectComponent.java]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 23]
```diff
@@ -134,8 +145,10 @@ public AbstractWidgetDescription caseSelectDescription(org.eclipse.sirius.compon
             return this.interpreter.evaluateExpression(variableManager.getVariables(), candidateExpression).asObjects().orElse(new ArrayList<>());
         };
         BiFunction<VariableManager, String, IStatus> selectNewValueHandler = this.getSelectNewValueHandler(viewSelectDescription.getBody());
+        SelectDescriptionStyle selectDescriptionStyle = viewSelectDescription.getStyle();
+
         // @formatter:off
-        return SelectDescription.newSelectDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.SelectDescription.Builder selectBuilder = SelectDescription.newSelectDescription(descriptionId)
```
Comment (gcoutable) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:151]: No need for the qualified name

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 24]
```diff
@@ -157,17 +176,24 @@ public AbstractWidgetDescription caseTextAreaDescription(TextAreaDescription tex
         StringValueProvider labelProvider = this.getStringValueProvider(textAreaDescription.getLabelExpression());
         StringValueProvider valueProvider = this.getStringValueProvider(textAreaDescription.getValueExpression());
         BiFunction<VariableManager, String, IStatus> newValueHandler = this.getNewValueHandler(textAreaDescription.getBody());
+        TextareaDescriptionStyle textareaDescriptionStyle = textAreaDescription.getStyle();
 
         // @formatter:off
-        return TextareaDescription.newTextareaDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.TextareaDescription.Builder textareaBuilder = TextareaDescription.newTextareaDescription(descriptionId)
```
Comment (gcoutable) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:182]: No need for the qualified name

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 25]
```diff
@@ -184,8 +210,10 @@ public AbstractWidgetDescription caseMultiSelectDescription(org.eclipse.sirius.c
             return this.interpreter.evaluateExpression(variableManager.getVariables(), candidateExpression).asObjects().orElse(new ArrayList<>());
         };
         BiFunction<VariableManager, List<String>, IStatus> multiSelectNewValueHandler = this.getMultiSelectNewValuesHandler(multiSelectDescription.getBody());
+        MultiSelectDescriptionStyle multiSelectDescriptionStyle = multiSelectDescription.getStyle();
+
         // @formatter:off
-        return MultiSelectDescription.newMultiSelectDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.MultiSelectDescription.Builder multiSelectBuilder = MultiSelectDescription.newMultiSelectDescription(descriptionId)
```
Comment (gcoutable) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:216]: No need for the qualified name

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 26]
```diff
@@ -220,8 +254,10 @@ public AbstractWidgetDescription caseRadioDescription(org.eclipse.sirius.compone
         };
 
         BiFunction<VariableManager, String, IStatus> newValueHandler = this.getSelectNewValueHandler(radioDescription.getBody());
+        RadioDescriptionStyle radioDescriptionStyle = radioDescription.getStyle();
+
         // @formatter:off
-        return RadioDescription.newRadioDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.RadioDescription.Builder radioBuilder = RadioDescription.newRadioDescription(descriptionId)
```
Comment (gcoutable) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:260]: No need for the qualified name

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/CheckboxElementProps.java [hunk 27]
```diff
@@ -123,6 +137,7 @@ public CheckboxElementProps build() {
             checkboxElementProps.label = Objects.requireNonNull(this.label);
             checkboxElementProps.value = Objects.requireNonNull(this.value);
             checkboxElementProps.newValueHandler = Objects.requireNonNull(this.newValueHandler);
+            checkboxElementProps.style = this.style;
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/CheckboxElementProps.java:140]: Optional on purpose

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/TextareaElementProps.java [hunk 28]
```diff
@@ -123,6 +137,7 @@ public TextareaElementProps build() {
             textareaElementProps.label = Objects.requireNonNull(this.label);
             textareaElementProps.value = Objects.requireNonNull(this.value);
             textareaElementProps.newValueHandler = Objects.requireNonNull(this.newValueHandler);
+            textareaElementProps.style = this.style;
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/TextareaElementProps.java:140]: Optional on purpose

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 29]
```diff
@@ -129,12 +128,17 @@ private Checkbox instantiateCheckbox(CheckboxElementProps props, List<Object> ch
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        return Checkbox.newCheckbox(props.getId())
+        org.eclipse.sirius.components.forms.Checkbox.Builder checkboxBuilder = Checkbox.newCheckbox(props.getId())
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:131]: No need for the qualified name

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 30]
```diff
@@ -154,61 +158,81 @@ private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
...
+        return radioBuilder.build();
         // @formatter:on
     }
 
     private Select instantiateSelect(SelectElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        return Select.newSelect(props.getId())
+        org.eclipse.sirius.components.forms.Select.Builder selectBuilder = Select.newSelect(props.getId())
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:179]: No need for the qualified name

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 31]
```diff
@@ -154,61 +158,81 @@ private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
...
+        return selectBuilder.build();
         // @formatter:on
     }
 
     private MultiSelect instantiateMultiSelect(MultiSelectElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        return MultiSelect.newMultiSelect(props.getId())
+        org.eclipse.sirius.components.forms.MultiSelect.Builder multiSelectBuilder = MultiSelect.newMultiSelect(props.getId())
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:198]: No need for the qualified name

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 32]
```diff
@@ -154,61 +158,81 @@ private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
...
+        return multiSelectBuilder.build();
         // @formatter:on
     }
 
     private Textarea instantiateTextarea(TextareaElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        return Textarea.newTextarea(props.getId())
+        org.eclipse.sirius.components.forms.Textarea.Builder textareaBuilder = Textarea.newTextarea(props.getId())
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:217]: No need for the qualified name

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 33]
```diff
@@ -154,61 +158,81 @@ private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
...
+        return textareaBuilder.build();
         // @formatter:on
     }
 
     private Textfield instantiateTextfield(TextfieldElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        Builder textfieldBuilder = Textfield.newTextfield(props.getId())
+        org.eclipse.sirius.components.forms.Textfield.Builder textfieldBuilder = Textfield.newTextfield(props.getId())
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:235]: No need for the qualified name

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 34]
```diff
@@ -81,7 +85,7 @@ public AbstractWidgetDescription caseTextfieldDescription(org.eclipse.sirius.com
         TextfieldDescriptionStyle textfieldDescriptionStyle = viewTextfieldDescription.getStyle();
 
         // @formatter:off
-        Builder textfieldBuilder = TextfieldDescription.newTextfieldDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.TextfieldDescription.Builder textfieldBuilder = TextfieldDescription.newTextfieldDescription(descriptionId)
```
Comment (gcoutable) [inline backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:88]: No need for the qualified name

**Review by gcoutable — CHANGES_REQUESTED** [review]
I take it from here

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java [hunk 35]
```diff
@@ -107,17 +111,24 @@ public AbstractWidgetDescription caseCheckboxDescription(org.eclipse.sirius.comp
         String valueExpression = Optional.ofNullable(viewCheckboxDescription.getValueExpression()).orElse(""); //$NON-NLS-1$
         BooleanValueProvider valueProvider = new BooleanValueProvider(this.interpreter, valueExpression);
         BiFunction<VariableManager, Boolean, IStatus> newValueHandler = this.getNewValueHandler(viewCheckboxDescription.getBody());
+        CheckboxDescriptionStyle checkboxDescriptionStyle = viewCheckboxDescription.getStyle();
 
         // @formatter:off
-        return CheckboxDescription.newCheckboxDescription(descriptionId)
+        org.eclipse.sirius.components.forms.description.CheckboxDescription.Builder checkboxBuilder = CheckboxDescription.newCheckboxDescription(descriptionId)
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:117]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java (hunk 24 above)
Comment (gcoutable) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:182]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java (hunk 25 above)
Comment (gcoutable) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:216]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java (hunk 26 above)
Comment (gcoutable) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:260]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/CheckboxStyle.java [hunk 36]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public Builder color(String color) {
+            this.color = Objects.requireNonNull(color);
+            return this;
+        }
+
+        public CheckboxStyle build() {
+            CheckboxStyle checkboxStyle = new CheckboxStyle();
+            checkboxStyle.color = this.color;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/CheckboxStyle.java:67]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/MultiSelectElementProps.java [hunk 37]
```diff
@@ -140,6 +154,7 @@ public MultiSelectElementProps build() {
             multiSelectElementProps.options = Objects.requireNonNull(this.options);
             multiSelectElementProps.values = List.copyOf(this.values);
             multiSelectElementProps.newValuesHandler = Objects.requireNonNull(this.newValuesHandler);
+            multiSelectElementProps.style = this.style;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/MultiSelectElementProps.java:157]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/RadioElementProps.java [hunk 38]
```diff
@@ -124,6 +138,7 @@ public RadioElementProps build() {
             radioElementProps.label = Objects.requireNonNull(this.label);
             radioElementProps.options = Objects.requireNonNull(this.options);
             radioElementProps.newValueHandler = Objects.requireNonNull(this.newValueHandler);
+            radioElementProps.style = this.style;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/elements/RadioElementProps.java:141]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java [hunk 39]
```diff
@@ -154,61 +158,81 @@ private Radio instantiateRadio(RadioElementProps props, List<Object> children) {
         List<Diagnostic> diagnostics = this.getDiagnosticsFromChildren(children);
 
         // @formatter:off
-        return Radio.newRadio(props.getId())
+        org.eclipse.sirius.components.forms.Radio.Builder radioBuilder = Radio.newRadio(props.getId())
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:161]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java (hunk 31 above)
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:198]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java (hunk 33 above)
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/renderer/FormElementFactory.java:235]: Done

File: backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java (hunk 34 above)
Comment (gcoutable) [reply backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/form/ViewFormDescriptionConverterSwitch.java:88]: Done

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/CheckboxDescription.java [hunk 40]
```diff
@@ -91,6 +98,8 @@ public static final class Builder {
 
         private Function<Object, String> messageProvider;
 
+        private Function<VariableManager, CheckboxStyle> styleProvider = vm -> null;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/CheckboxDescription.java:101]: It's ok, seen with @sbegaudeau

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/SelectDescription.java [hunk 41]
```diff
@@ -115,6 +122,8 @@ public static final class Builder {
 
         private Function<Object, String> messageProvider;
 
+        private Function<VariableManager, SelectStyle> styleProvider = vm -> null;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/SelectDescription.java:125]: It's ok, seen with @sbegaudeau

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/TextareaDescription.java [hunk 42]
```diff
@@ -91,6 +98,8 @@ public static final class Builder {
 
         private Function<Object, String> messageProvider;
 
+        private Function<VariableManager, TextareaStyle> styleProvider = vm -> null;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/TextareaDescription.java:101]: It's ok, seen with @sbegaudeau

**Review by gcoutable — APPROVED** [review]
I will merge it after the build has finished.

**Pull request merged**